tests/utils/get-related-pages-with-images.test.js

```diff
@@ -0,0 +1,36 @@
+import { getRelatedPagesWithImages } from '../../src/utils/setup/get-related-pages-with-images';
+
+it('should get pages and images for posts related to an article', () => {
+    const articleData = {
+        '/foo': {
+            query_fields: {
+                'atf-image': 'IMAGE_FILE',
+            },
+        },
+        '/bar': {
+            query_fields: {},
+        },
+    };
+    const blankArticle = {
+        query_fields: {},
+    };
+    const articleWithRelatedImage = {
+        query_fields: {
+            related: [{ target: '/foo' }],
+        },
+    };
+    const articleWithoutRelatedImage = {
+        query_fields: {
+            related: [{ target: '/bar' }],
+        },
+    };
+
+    expect(getRelatedPagesWithImages(blankArticle, articleData)).toEqual([]);
+    expect(
+        getRelatedPagesWithImages(articleWithRelatedImage, articleData)[0].image
+    ).toBe('IMAGE_FILE');
```

Comment:
can we replace `'IMAGE_FILE' with `arricleData['/foo'].query_fields['atf-image']` for more clarity?